### PR TITLE
Log server computes merkle hash at queue time.

### PR DESCRIPTION
**Pull request by AlCutter**

### CODE REVIEW

File: server/log_rpc_server_test.go [hunk 1]
```diff
@@ -9,22 +9,29 @@ import (
...
-var leaf1 = trillian.LogLeaf{LeafIndex: 1, MerkleLeafHash: []byte("hash"), LeafValue: []byte("value"), ExtraData: []byte("extra")}
-var leaf3 = trillian.LogLeaf{LeafIndex: 3, MerkleLeafHash: []byte("hash3"), LeafValue: []byte("value3"), ExtraData: []byte("extra3")}
-var expectedLeaf1 = trillian.LogLeaf{LeafIndex: 1, MerkleLeafHash: []byte("hash"), LeafValue: []byte("value"), ExtraData: []byte("extra")}
-var expectedLeaf3 = trillian.LogLeaf{LeafIndex: 3, MerkleLeafHash: []byte("hash3"), LeafValue: []byte("value3"), ExtraData: []byte("extra3")}
+var leaf1Data = []byte("value")
+var leaf3Data = []byte("value3")
+
+var leaf1 = trillian.LogLeaf{LeafIndex: 1, MerkleLeafHash: th.HashLeaf(leaf1Data), LeafValue: leaf1Data, ExtraData: []byte("extra")}
+var leaf3 = trillian.LogLeaf{LeafIndex: 3, MerkleLeafHash: th.HashLeaf(leaf3Data), LeafValue: leaf3Data, ExtraData: []byte("extra3")}
+var expectedLeaf1 = trillian.LogLeaf{LeafIndex: 1, MerkleLeafHash: th.HashLeaf(leaf1Data), LeafValue: leaf1Data, ExtraData: []byte("extra")}
```
Comment (codingllama) [inline server/log_rpc_server_test.go]: Is there any difference between leaf1 and expectedLeaf1? Could we do "expectedLeaf1 = leaf1"?

If not, could we do:
var expectedLeaf1 = trillian.LogLeaf{LeafIndex: leaf1.LeafIndex, MerkleLeafHash: leaf1.MerkleLeafHash, LeafValue: leaf1.LeafValue, ExtraData: leaf1.ExtraData}
?

(Ditto for leaf3 / expectedLeaf3.)

**Comment by daviddrysdale** [comment]
Don't you also need to update `mysql/log_storage.go` to retrieve and fill in the `MerkleLeafHash` in `dequeueLeaves` ?

File: server/log_rpc_server_test.go (hunk 1 above)
Comment (AlCutter) [inline server/log_rpc_server_test.go]: There doesn't seem to be any difference, and to be honest I don't know where there are two structs there so I've just deleted one of them.

**Comment by AlCutter** [comment]
@daviddrysdale , yup - thanks.

**Pull request merged**